lib/browser/objects-registry.js

```diff
@@ -28,11 +28,9 @@ class ObjectsRegistry {
       owner = this.owners[webContentsId] = new Set()
       this.registerDeleteListener(webContents, webContentsId)
     }
-    if (!owner.has(id)) {
```

Comment:
Both `remote` and `ObjectsRegistry` assume that one `webContents` can only hold one reference to an object in the main process, so when `remote` requests the same object for multiple times in one renderer process, the reference count to the object only adds one. The `if (!owner.has(id))` condition is to guard this assumption.

If we change this, the object would be leaked in the main process when a `webContents` requested the same remote object for multiple times.

Regarding the issue #9387:

```
initial load
hello.js was added to objects-registry and referenced count increased
reload
hello.js was added to objects-registry(no reference count increased)
```

The root cause is: after page was reloaded all remote objects referenced by the reloaded `webContents` should be cleared, but for some reason it is not true for sandboxed `webContents`.

The clearance is done by listening to the `render-view-deleted` event of `webContents` in the `registerDeleteListener` method, the bug is probably caused by the `render-view-deleted` event not emitting when reloading sandboxed `webContents`.